google_analytics/aggregation.py

```diff
@@ -1,7 +1,138 @@
+import json
 import os
+import shutil
+import glob
+import re
 import warnings
-import numpy as np
+
 import pandas as pd
+import numpy as np
+
+
+def reduce_df(path, output, nrows=None, chunksize=20000):
+    """ Load Google analytics data from JSON into a Pandas.DataFrame. """
+    if nrows and chunksize:
+        msg = "Reading {} rows in chunks of {}. We are gonna need {} chunks"
+        print(msg.format(nrows, chunksize, nrows / chunksize))
+
+    temp_dir = "../data/temp"
+    if not os.path.exists(temp_dir):
+        os.makedirs(temp_dir)
+
+    JSON_COLUMNS = ['device', 'geoNetwork', 'totals', 'trafficSource']
+
+    i = 0
+    for chunk in pd.read_csv(path,
+                             converters={column: json.loads for column in JSON_COLUMNS},
+                             dtype={'fullVisitorId': 'str'},
+                             nrows=nrows,
+                             chunksize=chunksize):
+
+        chunk = chunk.reset_index()
+
+        # Normalize JSON columns
+        for column in JSON_COLUMNS:
+            column_as_df = pd.io.json.json_normalize(chunk[column])
+            chunk = chunk.drop(column, axis=1).merge(column_as_df, right_index=True, left_index=True)
+
+        # Parse date
+        chunk['date'] = chunk['date'].apply(lambda x: pd.datetime.strptime(str(x), '%Y%m%d'))
+
+        # Only keep relevant columns
+        cols = ['date', 'fullVisitorId', 'operatingSystem', 'country', 'browser',
```

Comment:
yeap we should

Reply:
but then how would we aggregate? I mean a visitor can have multiple `visitStartTime` values, one for each section. Perhaps aggregate by concatenation so that the grouped field is a list of all `visitStartTime`s?